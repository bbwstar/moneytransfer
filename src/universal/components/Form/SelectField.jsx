import React from 'react';
import PropTypes from 'prop-types';

const SelectField = props =>
  (<div className="field">
    <label className="label" htmlFor={props.name}>
      {props.nameFormattedMessage}
    </label>
    <p className="control">
      <span className="select is-medium is-fullwidth">
        <select
          name={props.name.toLowerCase()}
          value={props.value}
          onChange={props.handleChange}
          id={props.name.toLowerCase()}
        >
          <option />
          {props.options.map(option =>
            (<option key={option.name} value={option.value ? option.value : option.name}>
              {option.name}
            </option>),
          )}
        </select>
      </span>
    </p>
  </div>);

SelectField.propTypes = {
  name: PropTypes.string.isRequired,
  nameFormattedMessage: PropTypes.element.isRequired,
  value: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      value: PropTypes.string,
    }),
  ).isRequired,
};

export default SelectField;
