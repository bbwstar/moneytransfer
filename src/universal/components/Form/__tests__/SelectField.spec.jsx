import React from 'react';
import { FormattedMessage } from 'react-intl';
import toJson from 'enzyme-to-json';
import { mountWithIntl } from 'utils/intl-enzyme-test-helper';

import SelectField from '../SelectField';

const handleChange = jest.fn();
const options = [{ name: 'First Option', value: 'First Value' }, { name: 'Second Option' }];
const wrapper = mountWithIntl(
  <SelectField
    name="test"
    nameFormattedMessage={<FormattedMessage id="testMessage" defaultMessage="testMessage" />}
    value="defaultValue"
    handleChange={handleChange}
    options={options}
  />,
);

describe('TextField (Snapshot)', () => {
  it('renders without exploding', () => {
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
  });

  it('calls handleChange function on input change', () => {
    wrapper.find('select').simulate('change');

    expect(handleChange).toBeCalled();
  });
});
