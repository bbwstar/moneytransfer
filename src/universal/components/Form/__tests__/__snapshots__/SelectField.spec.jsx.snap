// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField (Snapshot) renders without exploding 1`] = `
<SelectField
  handleChange={[Function]}
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": Object {},
      "now": [Function],
      "textComponent": "span",
    }
  }
  name="test"
  nameFormattedMessage={
    <FormattedMessage
      defaultMessage="testMessage"
      id="testMessage"
      values={Object {}}
    />
  }
  options={
    Array [
      Object {
        "name": "First Option",
        "value": "First Value",
      },
      Object {
        "name": "Second Option",
      },
    ]
  }
  value="defaultValue"
>
  <div
    className="field"
  >
    <label
      className="label"
      htmlFor="test"
    >
      <FormattedMessage
        defaultMessage="testMessage"
        id="testMessage"
        values={Object {}}
      >
        <span>
          testMessage
        </span>
      </FormattedMessage>
    </label>
    <p
      className="control"
    >
      <span
        className="select is-medium is-fullwidth"
      >
        <select
          id="test"
          name="test"
          onChange={[Function]}
          value="defaultValue"
        >
          <option />
          <option
            value="First Value"
          >
            First Option
          </option>
          <option
            value="Second Option"
          >
            Second Option
          </option>
        </select>
      </span>
    </p>
  </div>
</SelectField>
`;
